### PR TITLE
[CAT-2666] - Refactor usages of Map to SparseArray

### PR DESCRIPTION
**Pull request by Hydro8182**
I left in the usage of HashMap in ScratchConversionManager because it's used for a synchronizedMap there.

### CODE REVIEW

**Comment by 84n4n4** [comment]
test this please

File: catroid/src/main/java/org/catrobat/catroid/scratchconverter/protocol/Job.java [hunk 1]
```diff
@@ -48,8 +45,7 @@
 
 		private int state;
 
-		@SuppressLint("UseSparseArrays")
-		private static Map<Integer, State> map = new HashMap<>();
+		private static SparseArray<State> map = new SparseArray<>();
```
Comment (84n4n4) [inline catroid/src/main/java/org/catrobat/catroid/scratchconverter/protocol/Job.java]: can you please rename the "map" to something better. eg. "states"

**Comment by Hydro8182** [comment]
Done

**Comment by 84n4n4** [comment]
test this please

**Comment by 84n4n4** [comment]
please run espresso tests

**Pull request merged**